File: kotlin/src/main/kotlin/128LongestConsecutiveSequence.kt
import org.junit.Assert.assertEquals
import org.junit.Test

//https://leetcode.com/problems/longest-consecutive-sequence/

class LongestConsecutiveSequence{
  fun longestConsecutive(nums: IntArray): Int {
    val remain = nums.toHashSet()
    var result = 0
    while(remain.isNotEmpty()){
      val v = remain.first()
      remain.remove(v)
      var count = 1
      for(dir in listOf(-1, 1)){
        var next = v+dir
        while(remain.contains(next)){
          count ++
          remain.remove(next)
          next += dir
        }
      }
      result = maxOf(result, count)
    }

    return result
  }

  @Test
  fun test1(){
    assertEquals(4, longestConsecutive(intArrayOf(100, 4, 200, 1, 3, 2)))
  }
}

File: python/128_LongestConsecutiveSequence.py
from unittest import TestCase
# https://leetcode.com/problems/longest-consecutive-sequence/


class LongestConsecutiveSequence(TestCase):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        remain = set(nums)
        result = 0
        while remain:
            num = remain.pop()
            count = 1
            for dir in (-1, 1):
                v = num+dir
                while v in remain:
                    remain.remove(v)
                    count += 1
                    v += dir
            result = max(result, count)
        return result

    def test1(self):
        self.assertEqual(4, self.longestConsecutive([100, 4, 200, 1, 3, 2]))
